src/victory-primitives/bar.js

```diff
@@ -81,7 +81,8 @@ export default class Bar extends React.Component {
   getBarWidth(props, style) {
     const {width, data} = props;
     const padding = props.padding.left || props.padding;
-    const defaultWidth = data.length === 0 ? 8 : (width - 2 * padding) / data.length;
+    const length = Array.isArray(data) ? data.length : 0;
```

Comment:
It'd take a little more digging, but it might be better if we figure out where the possibility of `null` data is coming from. We already have a `Data` module that returns an array of data, empty or not - if we find the source of `null` here and enforce the same constraints on it, we won't have to do these type checks internally.